lib/unification.h

```diff
@@ -1,52 +1,123 @@
 #pragma once
+// This is the unification c++ algorithm
 
-#include "fol/Constant.h"
-#include "fol/Variable.h"
-#include "util.h"
 #include <iostream>
-#include <optional>
-#include <unordered_map>
+#include <vector>
+#include <stack>
```

Comment:
Do we need to include the `<stack>` header?

Reply:
Nope!